test/Hpack/ConfigSpec.hs

```diff
@@ -1151,3 +1151,25 @@ spec = do
         it "returns an error" $ \dir -> do
           let file = dir </> "package.yaml"
           readPackageConfig file `shouldReturn` Left [i|#{file}: Yaml file not found: #{file}|]
+
+      context "when included yaml file does not exist" $ do
+        it "returns an error" $ \dir -> do
+          let file = dir </> "package.yaml"
+          let included = "authors.yaml"
+          let includedAbs = dir </> included
+          writeFile file [i|
+            author: !include "#{included}"
+            |]
+          readPackageConfig file `shouldReturn` Left [i|#{file}: Yaml file not found: #{includedAbs}|]
+
+    it "accepts !include directive" $ do
+      let included = "authors.yaml"
+      withPackageConfig [i|
+        author: !include "#{included}"
+        |]
+        (writeFile included [i|
+          - A1
+          - A2
+        |]
+        )
+        (packageAuthor >>> (`shouldBe` ["A1", "A2"]))
```

Comment:
"Interesting" article. Too bad it offers no alternatives. Not having integration tests at all and just having it blow up in production is, in my humble opinion, not a better alternative. :-)

But I can agree that slow tests are really bad, but I think that's a technical problem, not a fundamental one. For a long time I wanted qemu to support superfast copy on write snapshots via forking (with readonly file/disk-backed storage, obviously) which I think could help a lot, but never had enough time/budget to implement that.

Anyway, I dropped the two tests.